src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -17631,5 +17744,420 @@ public void RefFieldTypeAllowsRefLike()
                 Diagnostic(ErrorCode.ERR_BindToBogus, "F").WithArguments("R2.F").WithLocation(6, 12)
                 );
         }
+
+        [Fact]
+        public void RestrictedTypesInRecords()
+        {
+            var src = @"
+record C<T>(
+    T P1
+    ) where T : allows ref struct;
+";
+
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyEmitDiagnostics(
+                // (3,5): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     T P1
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(3, 5)
+                );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void AnonymousDelegateType_01_ActionDisallowsRefStruct(bool s2IsRefStruct)
```

Comment:
Consider testing a scenario where we infer delegate type from a method with a parameter that has a default value (so we are forced to synthesize a delegate type for that) - the resulting synthesized delegate type should have `T : allows ref struct` even if it's not used with a ref struct or "allows ref struct" type parameter. (And it should not have that constraint on Net7.)

Reply:
> Consider testing a scenario where we infer delegate type from a method with a parameter that has a default value (so we are forced to synthesize a delegate type for that) - the resulting synthesized delegate type should have `T : allows ref struct` even if it's not used with a ref struct or "allows ref struct" type parameter. (And it should not have that constraint on Net7.)

I do not find this scenario interesting. Constraints on type parameters do not depend on specific type arguments used for  specific call sites. Effects of default values and specific types involved into specific signature are not related to each other. In my opinion the added tests adequately cover affected code paths and various flavors of generic anonymous delegate types. Please let me know if you disagree.

Reply:
BTW, there is a test with a pointer type, which is not a `ref` struct.